frontend/fingertips-frontend/components/views/ViewsContext.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { SearchParams, SearchStateParams } from '@/lib/searchStateManager';
+import { ViewsContext } from './ViewsContext';
+import { render } from '@testing-library/react';
+import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
+
+const mockOneIndicatorOneAreaView = jest.fn();
+jest.mock(
+  './OneIndicatorOneAreaView/',
+  () =>
+    function fn() {
+      mockOneIndicatorOneAreaView();
+      return <div />;
+    }
+);
+
+const mockOneIndicatorTwoOrMoreAreasView = jest.fn();
+jest.mock(
+  './OneIndicatorTwoOrMoreAreasView/',
+  () =>
+    function fn() {
+      mockOneIndicatorTwoOrMoreAreasView();
+      return <div />;
+    }
+);
+
+const mockTwoOrMoreIndicatorsAreasView = jest.fn();
+jest.mock(
+  './TwoOrMoreIndicatorsAreasView/',
+  () =>
+    function fn() {
+      mockTwoOrMoreIndicatorsAreasView();
+      return <div />;
+    }
+);
+
+const mockTwoOrMoreIndicatorsEnglandView = jest.fn();
+jest.mock(
+  './TwoOrMoreIndicatorsEnglandView/',
+  () =>
+    function fn() {
+      mockTwoOrMoreIndicatorsEnglandView();
+      return <div />;
+    }
+);
+
+async function generateSearchParams(value: SearchStateParams) {
+  return value;
+}
+
+describe('ViewsContext', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it.each([
+    [['foo'], ['zam'], mockOneIndicatorOneAreaView],
+    [['foo', 'bar'], ['zam'], mockOneIndicatorTwoOrMoreAreasView],
+    [['foo'], ['foo', 'bar', 'baz'], mockTwoOrMoreIndicatorsAreasView],
+    [[areaCodeForEngland], ['foo', 'bar'], mockTwoOrMoreIndicatorsEnglandView],
+  ])(
+    'should return the expected view',
+    async (areaCodes, indicators, correctView) => {
```

Comment:
Given your views are named with indicators first and then areas, I think this test would be easier to follow if you swapped the order of the parameters here:
```suggestion
    async (indicators, areaCodes, correctView) => {
```

Reply:
also switched the test data to match